src/Controller/StoreApiController.php

```diff
@@ -302,6 +327,33 @@ public function postPaymentDetails(
             return new JsonResponse($message, 500);
         }
 
+        // If donation token is present in the result, store it in the custom fields of order transaction.
+        $donationToken = $result->getDonationToken();
+        if (!is_null($donationToken) &&
+            $this->configurationService->isAdyenGivingEnabled($context->getSalesChannelId())) {
+            $storedTransactionCustomFields = $paymentResponse->getOrderTransaction()->getCustomFields() ?: [];
+            $transactionCustomFields[PaymentResponseHandler::DONATION_TOKEN] = $donationToken;
+
+            $customFields = array_merge(
+                $storedTransactionCustomFields,
+                $transactionCustomFields
+            );
+
+            $paymentResponse->getOrderTransaction()->setCustomFields($customFields);
+            $orderTransactionId = $paymentResponse->getOrderTransactionId();
+            $context->getContext()->scope(
+                Context::SYSTEM_SCOPE,
+                function (Context $context) use ($orderTransactionId, $customFields) {
+                    $this->orderTransactionRepository->update([
+                        [
+                            'id' => $orderTransactionId,
+                            'customFields' => $customFields,
+                        ]
+                    ], $context);
+                }
+            );
+        }
+
```

Comment:
this logic is a duplicate, since this is already done in `\Adyen\Shopware\Handlers\PaymentResponseHandler::handlePaymentResponse` which is called from `\Adyen\Shopware\Service\PaymentDetailsService::getPaymentDetails` which is called above

Reply:
Hello @peterojo, thank you for your review. This block is for saving donation tokens after the 3DS payments. `handlePaymentResponse` doesn't have donation token information before the 3DS auth. After the authentication, we are not calling `handleShopwareApis()` which is responsible for saving custom fields. Instead, we are calling `handleAdyenApis()` where we can't save custom fields.